File: Python/reverse_string.py
# Display the characters in a string in reverse order

# define test case:
string = 'abcdefg';


# Fastest & easiest route utilizes extended slice to copy in reverse
# leaving start/end blank copies whole string
#
print string[::-1]


# Can use built-in reverse function and concatenation:
#
print ''.join(reversed(string))


# Iterative:
#
def reverse_string(string):
	reversed = ''
	for characters in string:
	  reversed = characters + reversed
	return reversed
print reverse_string(string)

File: PHP/reverse_string.php
<?php
/*
* Display the characters in a string in reverse order
* 
* Define test case:
*/
$str = 'abcdefg';
/*
*
* Ordinarily, we might just use 
*   echo strrev("abcdefg");
* to achieve the desired output.
*
* But today is no ordinary day! 
* Here is an iterative solution in PHP:
*
*
* Define length variable as length of string
* Iterate backwards through string, 
* outputting each character in reverse.
*/
function reverse_string($string){
  $length = strlen($string);

  for ($i = $length; $i >= 0; $i--) {
    echo $string[$i-1];
  }
}
/*
* Test output: gfedcba
*/
reverse_string($str);
?>
